spec.bs

```diff
@@ -2615,6 +2622,53 @@ a {{ReportingBrowserSignals}} |browserSignals|, and a [=direct from seller signa
      :: |reportingMacroMap|
 </div>
 
+<h3 id="canloadadauctionfencedframe">canLoadAdAuctionFencedFrame()</h3>
+
+*This first introductory paragraph is non-normative.*
+
+The process of running an ad auction through {{Window/navigator}}.{{Navigator/runAdAuction()}} is an
+expensive operation. To prevent wasted cycles if the resulting ad cannot be loaded in the current
+context, {{Window/navigator}}.{{Navigator/canLoadAdAuctionFencedFrame()}} is provided as a method to
+determine, before the ad auction begins, whether an ad auction-created <{fencedframe}> is allowed to
+be loaded in the current context. A <{fencedframe}> has restrictions on where and how it can be
+loaded that do not exist with <{iframe}>s.
+
+<xmp class="idl">
+[SecureContext]
+partial interface Navigator {
+  boolean canLoadAdAuctionFencedFrame();
+};
+</xmp>
+
+The <dfn for=Navigator method>canLoadAdAuctionFencedFrame()</dfn> method steps are:
+
+1. Let |global| be [=this=]'s [=relevant global object=].
+
+1. If |global|'s [=Window/browsing context=]'s [=required CSP=] is not null, then return false.
+
+1. Let |CSPList| be [=this=]'s [=relevant settings object=]'s [=environment settings object/policy
+   container=]'s [=policy container/CSP list=].
+
+1. [=list/For each=] |policy| of |CSPList|:
+
+  1. [=set/For each=] |directive| of |policy|'s [=policy/directive set=]:
+
+    1. If |directive|'s [=directive name|name=] is either "`fenced-frame-src`", "`frame-src`",
+       "`child-src`", or "`default-src`", and if |directive|'s [=directive value|value=] does not
+       [=set/contain=] any of "`https:`", "`https://*:*`", or "`*`", then return false.
+
+1. Let |sandboxFlags| be |global|'s [=associated Document=]'s [=Document/active sandboxing flag
+   set=].
+
+1. [=list/For each=] |mandatoryUnsandboxedFlags| of TBD:
+
+   Issue: TODO: Get the mandatory unsandboxed flags list once it exists in the fenced frame spec.
+   (<a href="https://github.com/WICG/turtledove/issues/1206">WICG/turtledove#1206</a>)
+
+  1. If |sandboxFlags| [=set/contains=] |mandatoryUnsandboxedFlags|, then return false.

```

Comment:
I don't think the "contains" word can check if a __list__ is contained inside of another list, right? I think it only works for list items.

Reply:
The variable should've said `|mandatoryUnsandboxedFlag|`, and it will be a [sandboxing flag set](https://html.spec.whatwg.org/multipage/browsers.html#sandboxing-flag-set). I've fixed the typo and updated the comment to call out exactly what we're expecting `TBD` to be.

Reply:
ah right, subset is more correct. https://infra.spec.whatwg.org/#set-subset

Reply:
the comment was about the "contains" method cannot check if a set contains all elements of other set. It can only check if a set contains a certain single element. You may need to use https://infra.spec.whatwg.org/#set-subset to say "|mandatoryUnsandboxedFlags| is a [=set/subset=] of |sandboxFlags|", if |mandatoryUnsandboxedFlags| is also a set

Reply:
Sorry, I should've been more clear. `|mandatoryUnsandboxedFlag|` is a single flag in this case (I just mistyped the variable name), and `TBD` is the sandboxing flag **set**. But it looks like using [subset](https://infra.spec.whatwg.org/#set-subset) is a cleaner way to do this and won't require the loop.

Reply:
sorry, missed that.